### PR TITLE
*: add table storage info status api

### PR DESCRIPTION
**Pull request by jianzhiyao**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/4989

Problem Summary:
Add table storage infomation to status api.

### What is changed and how it works?

Proposal: https://github.com/pingcap/tidb/issues/4989

What's Changed:
new status api: http://127.0.0.1:10080/schema_storage

How it Works:
By querying INFORMATION_SCHEMA.TABLES.

### Check List 

Tests 

- Unit test

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM

### Release note 

- add table storage info status api

### CODE REVIEW

File: server/http_handler.go [hunk 1]
```diff
@@ -909,6 +915,128 @@ func (h flashReplicaHandler) handleStatusReport(w http.ResponseWriter, req *http
...
+	IndexLength   int64  `json:"index_length"`
+	DataFree      int64  `json:"data_free"`
+}
+
+func getSchemaTablesStorageInfo(h *schemaStorageHandler, schema *model.CIStr, table *model.CIStr) (messages []*schemaTableStorage, err error) {
+	var s session.Session
+	if s, err = session.CreateSession(h.Store); err != nil {
+		return
+	}
+	if s != nil {
```
Comment (AndreMouche) [inline server/http_handler.go:934]: it seems we do not need to do anything when s is nil, could we process this situation before, so we needn't always remember this situation until the end of the function.

File: server/http_handler.go [hunk 2]
```diff
@@ -909,6 +915,128 @@ func (h flashReplicaHandler) handleStatusReport(w http.ResponseWriter, req *http
...
+		defer s.Close()
+		if ctx, ok := s.(sessionctx.Context); ok {
```
Comment (AndreMouche) [inline server/http_handler.go:936]: ditto, less `{}`, return as early as possible.

**Review by AndreMouche — APPROVED** [review]
The logic LGTM, while I wish you could make this PR more friendly to the reviewers and maintainers, Please make the code as easy as possible, for example, it is a good habit if the number of  '{..{..{..}}}' less than 3.

File: server/http_handler.go (hunk 1 above)
Comment (jianzhiyao) [reply server/http_handler.go]: done

File: server/http_handler.go (hunk 2 above)
Comment (jianzhiyao) [reply server/http_handler.go]: done

**Review by AndreMouche — APPROVED** [review]
LGTM

**Comment by jianzhiyao** [comment]
PTAL@morgo

**Comment by jianzhiyao** [comment]
PTAL @morgo

**Review by morgo — APPROVED** [review]

**Comment by xhebox** [comment]
/merge cancel
CI fix at #25173

**Comment by xhebox** [comment]
/merge cancel
Looks like I fixed the panic, but now it goes into timeout.

**Comment by xhebox** [comment]
I'll take a look today.

**Review by AilinKid — APPROVED** [review]

**Comment by xhebox** [comment]
/merge cancel
So the timeout problem did not disappear. I'll take a look after my exam, i.e. next week.

**Comment by xhebox** [comment]
`TelemetryRotateWindowLoop` stuck the termination of the test. So yes, the test actually passed. But I can not reproduce the case on my laptop. Investigating the root cause...

**Pull request merged**